src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -83,8 +84,39 @@ public async IAsyncEnumerable<FeatureDefinition> GetAllFeatureDefinitionsAsync()
 
         private FeatureDefinition ReadFeatureDefinition(string featureName)
         {
-            IConfigurationSection configuration = GetFeatureDefinitionSections()
-                                                    .FirstOrDefault(section => section.Key.Equals(featureName, StringComparison.OrdinalIgnoreCase));
+            IConfigurationSection configuration;
+
+            if (!featureName.Contains(ConfigurationPath.KeyDelimiter))
+            {
+                configuration = GetFeatureDefinitionSections()
+                    .FirstOrDefault(section => section.Key.Equals(featureName, StringComparison.OrdinalIgnoreCase));
+            }
+            else
+            {
+                //
+                // Feature names with configuration path delimiters require traversing children sections for resolution
+                IEnumerable<string> sectionNames = featureName.Split(SectionSeparator, StringSplitOptions.None);
```

Comment:
Could we define `SectionSeparator` as a `string` instead of `string[]`? Another option is to directly use `ConfigurationPath.KeyDelimiter` because this is a special delimiter that's unlikely to change in .NET.

Reply:
I just stored it that way since `Split` accepts string[] and `ConfigurationPath.KeyDelimiter` is a string instead of a char.

Reply:
FWIW, [Split ](https://docs.microsoft.com/en-us/dotnet/api/system.string.split?view=net-6.0#System_String_Split_System_String_System_StringSplitOptions_)also accepts string delimiter.